src/toil/common.py

```diff
@@ -717,10 +746,35 @@ def getWorkflowDir(workflowID, configWorkDir=None):
             logger.info('Created the workflow directory at %s' % workflowDir)
         return workflowDir
 
-    def _assertContextManagerIsUsed(self):
-        # Assert that we are inside the context manager
-        if self.jobStore is None:
-            # __enter__() sets self.jobStore
+    def _runMainLoop(self, rootJob):
+        """
+        Runs the main loop with the given job.
+        :param toil.job.Job rootJob: The root job for the workflow.
+        :rtype: Any
+        """
+        with RealtimeLogger(self._batchSystem, level=self.options.logLevel if self.options.realTimeLogging else None):
+            # FIXME: common should not import from leader
+            from toil.leader import mainLoop
+            return mainLoop(config=self.config,
+                            batchSystem=self._batchSystem,
+                            provisioner=None,
+                            jobStore=self._jobStore,
+                            rootJobWrapper=rootJob,
+                            jobCache=self._jobCache)
+
+    def _shutdownBatchSystem(self):
+        """
+        Shuts down current batch system if it has been created.
+        """
+        assert self._batchSystem is not None
+
+        startTime = time.time()
+        logger.debug('Shutting down batch system')
+        self._batchSystem.shutdown()
+        logger.debug('Finished shutting down the batch system in %s seconds.' % (time.time() - startTime))
+
+    def _assertContextManagerUsed(self):
+        if not self._inContextManager:
             raise ToilContextManagerMisuseException("Toil class cannot be instantiated outside of a context manager")
```

Comment:
The message is misleading. An instance of the Toil class *is* a context manager. And Toil *can* be used without a `with` statement. It's just that certain methods of the Toil class can't be used outside the context manager.